storm-core/src/jvm/backtype/storm/scheduler/TopologyDetails.java

```diff
@@ -396,34 +410,50 @@ public void addResourcesForExec(ExecutorDetails exec, Map<String, Double> resour
             LOG.warn("Executor {} already exists...ResourceList: {}", exec, getTaskResourceReqList(exec));
             return;
         }
-        _resourceList.put(exec, resourceList);
+        this.resourceList.put(exec, resourceList);
     }
 
     /**
      * Add default resource requirements for a executor
      */
     public void addDefaultResforExec(ExecutorDetails exec) {
+        Double topologyComponentCpuPcorePercent = Utils.getDouble(this.topologyConf.get(Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT), null);
+        if (topologyComponentCpuPcorePercent == null) {
+            LOG.warn("default value for " + Config.TOPOLOGY_COMPONENT_CPU_PCORE_PERCENT + " needs to be set!");
+        }
```

Comment:
The config validation won't validate any configs that is not in the conf map.  Currently, there is no mechanism for checking for "must have" configs.  I do agree that we should also have a unit test to check this, which i will add, but I am for leaving the checks in here and throwing a run time exception if this happens